src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -89,7 +89,7 @@ protected override IMethodSymbol GenerateMethodDefinition(
                     attributes: [],
                     accessibility: Accessibility.Private,
                     modifiers: CreateMethodModifiers(),
-                    returnType: AnalyzerResult.ReturnType,
+                    returnType: this.GetFinalReturnType(),
```

Comment:
only when we're actually generating the final method declaration do we get teh type, potentially wrapped with 'Task'.  Everything else sees the 'core' type and can ask questions of that.